test/NuGet.Core.Tests/NuGet.Packaging.Test/MissingReadmeRuleTests.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NuGet.Commands;
+using NuGet.Common;
+using NuGet.Packaging.Rules;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.Packaging.Test
+{
+    //Package authoring best practices are an ever evolving set of rules and guidelines.
+    //Those soft recommendations are non-breaking validation message. 
+    public class MissingReadmeRuleTests
+    {
+        [Theory]
+        [InlineData("<readme>readme.md</readme>", true)]
+        [InlineData("", false)]
+        public void Validate_NuSpecFileWithReadme_GeneratesMessage(string readmeMetadata, bool readmeMetadataExists)
+        {
+            // Arrange
+            var nuspecContent = "<?xml version=\"1.0\" encoding=\"utf-8\"?>" +
+"<package xmlns=\"http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd\">" +
+"   <metadata>" +
+"        <id>test</id>" +
+"        <version>1.0.0</version>" +
+"        <authors>Unit Test</authors>" +
+"        <description>Sample Description</description>" +
+"        <language>en-US</language>" +
+readmeMetadata +
+"    <dependencies>" +
+"      <dependency id=\"System.Collections.Immutable\" version=\"4.3.0\" />" +
+"    </dependencies>" +
+"    </metadata>" +
+"</package>";
+
+            var readmeContent = "Test readme file.";
+
+            using (var testDirectory = TestDirectory.Create())
```

Comment:
You can probably just mock PackageArchiveReader and avoid having to do integration tests.

Reply:
Thanks for the info! Fixed~